test/rubygems/test_gem_package.rb

```diff
@@ -946,6 +946,87 @@ def test_verify_corrupt
     tf.close!
   end
 
+  def test_verify_corrupt_tar_metadata_entry
+    gem = tar_file_header("metadata.gz", "", 0, 999, Time.now)
```

Comment:
`verify` handles metadata, checksums, and data.tar.gz slightly differently, raising different errors when each is corrupt. That's why there's so many new tests here for seemingly the same thing.